### ISSUE TITLE
Add support for AOT in the wasm SDK

### ISSUE THREAD

**Issue by jeromelaban** [mono/mono#9825]
The Mono-wasm SDK currently uses the interpreter and this makes its performance pretty low (https://github.com/stefan-schweiger/dotWasmBenchmark), making the "giant switch" a hot target for performance (https://bugs.chromium.org/p/v8/issues/detail?id=7838#c7).

Enabling the AOT for the wasm sdk should theoretically provide a significant performance improvement, and make mono a viable target for demanding applications that rely on C# for rendering UI.

**Comment by stefan-schweiger** [comment]
Also see https://github.com/aspnet/Blazor/issues/26#issuecomment-400262994

> **Miguel de Icaza (2018-06-08 19:01):**
> We are working on it. What happened is this:
The interpreter is using “emscripten” that has a fairly complete libc that we can rely on. And our AOT engine was built on pure LLVM, which requires from us to write the full libc. The latter is the ideal place, because we get native linkers and immedite llvm support and so on. But woudl send us down the path of having to write that libc.
So for the short term, we are moving to emscripten the AOT compiler, make sure that we keep the compatibility. The downside though is that the emscripten tooling is older, so many of the better pieces like the LLVM linker are not available. So we are working through those things.
Basically, we had something done, but we had to start from scratch to work with what we have without forving us to write and emulate everything on our own. We could have attempted to blend those two, but that is also a major effort. So we think we can do emscripten for now via some artful hacks here and there.
